### PR TITLE
[IMP] general: iot printer troubleshooting

### PR DESCRIPTION
**Pull request by tiku-odoo**
Task: 
https://www.odoo.com/web#cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form&id=3853783

### CODE REVIEW

**Comment by tiku-odoo** [comment]
@lse-odoo 
@jero-odoo 

When you have a moment, can you review the changes proposed in lines: 346-404

Thanks, 
Tim

File: content/applications/general/iot/config/troubleshooting.rst
Comment (lse-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:346]: Note that I am not sure if it impacts other DYMO printers, might be better to be more generic and tell that the 450 DUO is the one that we support and on which it was tested on

File: content/applications/general/iot/config/troubleshooting.rst
Comment (lse-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:350]: technically false. If you check back the video it does not print at all if the raw driver is used (which is the default one set when cups detect the printer

File: content/applications/general/iot/config/troubleshooting.rst
Comment (lse-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:360]: same remark as the previous one, the printer is automatically seen with cups (as long as it is connected to the IoT box) & the "raw" driver is automatically given to it. You don't need to add it manually

File: content/applications/general/iot/config/troubleshooting.rst
Comment (lse-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:400]: there is no reason to do it (and I don't think it will change anything on the behavior)

File: content/applications/general/iot/config/troubleshooting.rst
Comment (jero-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:355]:
```suggestion
   The DYMO LabelWriter 450 DUO printer **must** already be connected and installed on the :abbr:`IoT
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (jero-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:362]:
```suggestion
From the :guilabel:`Local Printers` section, select the :guilabel:`DYMO LabelWriter 450 DUO Label
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (jero-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:369]:
```suggestion
On the following screen, modify the :guilabel:`Name` to something recognizable, as the original
```

**Review by lse-odoo — COMMENTED** [review]
I'm afraid that you are beside the point.

To summarize the video that I sent to you:

Issue:
When connectiing a DYMO LabelWriter 450, it will be recognised as an IoT device but will not print the label (the blue ligh blink but nothing happens)

-> By modifying the driver from "raw/raw (queue)" to "Dymo / Label writer 450" it will print but with a delay of ~10 seconds each time

The workaround that we found around that is to create a new printer using the existing one (see in the video the steps) and then it will print without any delay.
I don't know the actual reasons as there is seemingly no difference between the newly created and the "existing one" but it is the only solution that I found so far :/

File: content/applications/general/iot/config/troubleshooting.rst
Comment (jero-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:388]: I will defer to Sam on this, but since you are directing the user to click on the specific printer, this might need something other than italics? (possibly backticks?_

File: content/applications/general/iot/config/troubleshooting.rst
Comment (jero-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:402]:
```suggestion
To print a test label, click on the :guilabel:`Maintenance` drop-down menu to the left of the
```

**Review by jero-odoo — APPROVED** [review]
Hey @tiku-odoo, looks good to me! I had a few quick edits but I am approving this to go to the next round :+1:

File: content/applications/general/iot/config/troubleshooting.rst
Comment (tiku-odoo) [reply content/applications/general/iot/config/troubleshooting.rst:388]: @samueljlieber 

Thoughts on this?

Comment (samueljlieber) [reply content/applications/general/iot/config/troubleshooting.rst]: @tiku-odoo 
If the text "DYMO LabelWriter 450 DUO Label" is visible on the **Printers** page, then it needs a :guilabel:
```
Click into the :guilabel:`DYMO LabelWriter 450 DUO Label` printer...
```
If the text is not visible you can use italics to refer to the object the user clicks that represents the printer.

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:350]:
```suggestion
Things)` box, while using the label function. To resolve this issue, the printer must be reinstalled.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:352]:
```suggestion
homepage, and click :guilabel:`Add a Printer`.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:351]:
```suggestion
First, open the :guilabel:`OpenPrinting CUPS` administration page from the :abbr:`IoT (Internet of Things)` box
```
if this is how it appears on the interface, use this GUI -- if not, feel free to ignore this suggestion

**Comment by tiku-odoo** [comment]
@ksc-odoo 

This doc-change (lines 346-404) is ready for your review. The rest of this doc has been updated in the last 6-12 months. 

Thanks, 
Tim :+1: 

CC: @lse-odoo, I've made the changes you requested. The doc is much clearer now. Can you review to ensure my technical changes are acceptable?

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:389]:
```suggestion
All the printers installed on the OpenPrinting CUPS server appear, including the newly
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:390]: for both of this label mentions, I'd suggest using a GUI, especially if that's how they appear on the screen/UI

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:398]:
```suggestion
:guilabel:`Administration` drop-down menu, and select :guilabel:`Print Test Page`. The test label should
```

**Review by ksc-odoo — APPROVED** [review]
hey @tiku-odoo -- just finished my Final Review. Only have some minor adjustments that require your attention. When you make the necessary changes, go ahead and tag for Tech Review. Approving now 👍

**Comment by tiku-odoo** [comment]
@samueljlieber 

This doc is ready for your review. The changes added are in lines: 346-405

Thanks,
 Tim :+1:

**Review by lse-odoo — CHANGES_REQUESTED** [review]

File: content/applications/general/iot/config/troubleshooting.rst
Comment (tiku-odoo) [reply content/applications/general/iot/config/troubleshooting.rst:357]: @lse-odoo 

I've added this into the doc, can you review?

Thanks,
Tim

Comment (lse-odoo) [reply content/applications/general/iot/config/troubleshooting.rst:357]: in my opinion it is still misleading, the section itself is called "printer delay" so people for which it does not print at all will already skip the entire section. If you (and the other reviewer) think it is good like this then it's okay for me as you have more experience than me in documentation writing. But I don't really like the way it was modified compared to what I proposed. Maybe having a 3rd opinion from someone neuter who looked at the video can be useful ?

Comment (tiku-odoo) [reply content/applications/general/iot/config/troubleshooting.rst:356]: @lse-odoo 

I updated the process for you according to your video. Can you take another look before I bump this to final review again?

Thanks,
Tim :+1:

**Comment by lse-odoo** [comment]
@tiku-odoo great I prefer much more this version thank you for your continuous effort :) 

The content itself is good, but - personally - I think the title are not well suited. Wouldn't it be better to describe the issue themselves rather than the solution (with the paragraph before it introducing the problem).

If we check the TOC on the side it gives:
```
DYMO LabelWriter print issue
 > Set correct driver
 > Add new printer
```

which might not help a customer to solve this issue compared to:
```
DYMO LabelWriter print issue
 > DYMO LabelWriter does not print anything
 > DYMO LabelWriter print after ~10 seconds delay
```

**Comment by tiku-odoo** [comment]
@ksc-odoo 

The structure and copy of these changes were updated, can you re-look at lines: 
346-491

Thanks, 
Tim :+1:

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:351]:
```suggestion
In order to print anything, the correct :guilabel:`Make and Model` needs to be set, so the
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:362]:
```suggestion
   The DYMO LabelWriter 450 DUO printer contains two printers in one: a label printer and a
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:366]:
```suggestion
   To keep things consistent, both of the following processes detail the configuration for the DYMO
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:378]:
```suggestion
menu. Click into the printer in question, and select :guilabel:`Maintenance` in the first
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:408]:
```suggestion
setting the new driver, a confirmation page appears.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:423]:
```suggestion
prints out with a ten-second delay, if the driver update was successful.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:430]:
```suggestion
To reduce this delay, a new printer **must** be added. To add a new printer, follow the process below.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:435]:
```suggestion
To resolve the delay issue after modifying the driver, the printer **must** be reinstalled. To reinstall the printer, open
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:436]:
```suggestion
the OpenPrinting CUPS administration page by clicking :guilabel:`Printers server`, at the bottom of
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:437]:
```suggestion
the :abbr:`IoT (Internet of Things)` box homepage. Then, click on :guilabel:`Administration` in the
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:438]:
```suggestion
top menu, then click :guilabel:`Add a Printer`.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (ksc-odoo) [inline content/applications/general/iot/config/troubleshooting.rst:441]:
```suggestion
   If the DYMO LabelWriter 450 DUO printer is not printing at all, or is not recognizable (has a RAW
```

**Review by ksc-odoo — APPROVED** [review]
@tiku-odoo you're just about there! Only a handful of super small adjustments that require your attention. Approving now. Once those modifications are done, and this PR is ready to go, feel free to tag this for Tech Review! 👍

**Comment by tiku-odoo** [comment]
@samueljlieber 

This doc is ready for your review: 

Lines 346-491

Thanks, Tim :+1:

File: content/applications/general/iot/config/troubleshooting.rst
Comment (samueljlieber) [inline content/applications/general/iot/config/troubleshooting.rst:438]: Suggesting to use menuselection here since were directing the user through menus :)
```suggestion
::menuselection:`Printers server`, at the bottom of the :abbr:`IoT (Internet of Things)` box
homepage. Then, click on :menuselection:`Administration` in the top menu, then click :guilabel:`Add
a Printer`.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (samueljlieber) [inline content/applications/general/iot/config/troubleshooting.rst:450]: Just double checking that the same text outside the parenthesis are inside "(DYMO LabelWriter 450 DUO Label)", is this like that in the UI?

File: content/applications/general/iot/config/troubleshooting.rst
Comment (samueljlieber) [inline content/applications/general/iot/config/troubleshooting.rst:474]: Again suggesting to use menuselection when directing the user :)
```suggestion
:menuselection:`Printers` button in the top menu.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (samueljlieber) [inline content/applications/general/iot/config/troubleshooting.rst:379]: Menuselection :)
```suggestion
First, open the OpenPrinting CUPS console by clicking :menuselection:`Printers server` at the bottom
of the :abbr:`IoT (Internet of Things)` box homepage. Next, click on :menuselection:`Printers` in
the top menu. Click into the printer in question, and select :guilabel:`Maintenance` in the first
drop-down menu. Then, select :guilabel:`Modify Printer` in the second drop-down menu.
```

File: content/applications/general/iot/config/troubleshooting.rst
Comment (samueljlieber) [inline content/applications/general/iot/config/troubleshooting.rst:415]:
```suggestion
:menuselection:`Printers` button in the top menu.
```

**Review by samueljlieber — APPROVED** [review]
Hi @tiku-odoo, awesome job on the improvements to this doc. Your PR looks great to me, I only have a couple of nitpick suggestions to use menuselection when directing the user. This is a small suggestion, but I think it can help semantically– what do you think? 

Either way, I am approving now and delegating to you to merge at your discretion. Thanks and great job! 👍
...
@robodoo delegate=tiku-odoo

File: content/applications/general/iot/config/troubleshooting.rst
Comment (tiku-odoo) [reply content/applications/general/iot/config/troubleshooting.rst:450]: Yes this is in the UI

modify-select-printer

**Comment by tiku-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
